src/store/plugins/persistState.js

```diff
@@ -1,10 +1,10 @@
 import { cloneDeep } from 'lodash-es';
+import { detect } from 'detect-browser';
 
 const KEY = 'state';
 
-const setState = async (state) => {
-  await browser.storage.local.set({ [KEY]: cloneDeep(state) });
-};
+const setState = (state) =>
+  browser.storage.local.set({ [KEY]: detect().name === 'firefox' ? cloneDeep(state) : state });
```

Comment:
Is it only Firefox that is problematic and requires `cloneDeep` to be used? And why not keep it consistent across browsers and stick to using `cloneDeep` for all of them?

Reply:
> Is it only Firefox that is problematic and requires cloneDeep to be used?

I think so, also Firefox-based browsers can be problematic like Tor Browser, Pale Moon, and Waterfox

> And why not keep it consistent across browsers and stick to using cloneDeep for all of them?

Firstly, I want to have the explanation above be connected to git history. `cloneDeep` makes sense only for Firefox, also there needed not exactly the cloning, but proper evaluation of the Vuex state. Other browsers doing it by themselves, and without any mark (like firefox check) the current implementation looks like api misusage.

> Does cloneDeep introduce any performance drawbacks?

Nope.

Reply:
Sounds good. Thanks 👍